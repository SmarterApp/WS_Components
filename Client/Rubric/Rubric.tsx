import * as React from 'react'
import * as AboutItemModels from '../AboutItem/AboutItemModels';
import * as RubricEntry from './RubricEntry';
import * as Collapsible from './Collapsible';
import * as SampleResponse from '../SampleResponse/SampleResponse';

export class RubricComponent extends React.Component<AboutItemModels.Rubric, {}> {

    renderRubrics() {
        const rubricEntries = this.props.rubricEntries.map((re, i) => <RubricEntry.RubricEntryComponent {...re} key={String(i)} />);
        if (rubricEntries.length === 0) {
            return null;
        }

        let rubrics = (
            <div>
                <h4>Rubrics</h4>
                {rubricEntries}
            </div>
        );
        return rubrics;
    }

    renderSampleResponses() {
        let rubricSamples: JSX.Element[] = [];
        let i: number = 0;
    
        for (const sample of this.props.samples) {
            const key = `${i}:`;
            const responses = sample.sampleResponses.map((sr, idx) => <SampleResponse.SampleResponseComponent {...sr} key={key + String(idx)} />);
            rubricSamples.push(...responses);
            i++;
        }

        if (rubricSamples.length === 0) {
            return null;
        }

        let sampleResponses = (
            <div>
                <h4>Sample Responses</h4>
                {rubricSamples}
            </div>
        );

        return sampleResponses;
    }

    render() {
        const label = `${this.props.language} Rubric`;
        return (
            <Collapsible.CComponent label={label}>
                {this.renderRubrics()}
                {this.renderSampleResponses()}
            </Collapsible.CComponent>
        );
    }
}