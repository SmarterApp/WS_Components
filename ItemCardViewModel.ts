import * as GradeLevels from './GradeLevels'

export interface ItemCardViewModel {
    bankKey: number;
    itemKey: number;
    title: string;
    grade: GradeLevels.GradeLevels;
    gradeLabel: string;
    subjectCode: string;
    subjectLabel: string;
    claimCode: string;
    claimLabel: string;
    target: string;
    interactionTypeCode: string;
    interactionTypeLabel: string;
}