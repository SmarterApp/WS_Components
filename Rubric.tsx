import * as React from 'react';

export interface SampleResponse {
    purpose: string;
    scorePoint: string;
    name: string;
    sampleContent: string;
}

export interface RubricSample {
    maxValue: string;
    minValue: string;
    sampleResponses: SampleResponse[];
}

export interface RubricEntry {
    scorepoint: string;
    name: string;
    value: string;
}

export interface Rubric {
    language: string;
    rubricEntries: RubricEntry[];
    samples: RubricSample[];
}

export class RubricEntryComponent extends React.Component<RubricEntry, {}> {
    render() {
        const pointLabel = this.props.scorepoint === "1" ? "point" : "points";
        const label = `${this.props.name} (${this.props.scorepoint} ${pointLabel})`;
        return (
            <div dangerouslySetInnerHTML={{ __html: this.props.value }} />
        );
    }
}

export class SampleResponseComponent extends React.Component<SampleResponse, {}> {
    render() {
        const pointLabel = this.props.scorePoint == "1" ? "point" : "points";
        const label = `${this.props.name} (${this.props.scorePoint} ${pointLabel})`;
        return (
            <div className="sample-response">
                <b>Purpose: </b> {this.props.purpose} <br />
                <b>Sample Response: </b> <br />
                <div dangerouslySetInnerHTML={{ __html: this.props.sampleContent }} />
            </div>
        );
    }
}

export class RubricComponent extends React.Component<Rubric, {}> {

    renderRubrics() {
        const rubricEntries = this.props.rubricEntries.map((re, i) => <RubricEntryComponent {...re} key={String(i)} />);
        if (rubricEntries.length === 0) {
            return null;
        }

        let rubrics = (
            <div>
                <h4>Rubrics</h4>
                {rubricEntries}
            </div>
        );
        return rubrics;
    }

    renderSampleResponses() {
        let rubricSamples: JSX.Element[] = [];
        let i: number = 0;
        for (const sample of this.props.samples) {
            const key = `${i}:`;
            const responses = sample.sampleResponses.map((sr, idx) => <SampleResponseComponent {...sr} key={key + String(idx)} />);
            rubricSamples.push(...responses);
            i++;
        }
        if (rubricSamples.length === 0) {
            return null;
        }

        let sampleResponses = (
            <div>
                <h4>Sample Responses</h4>
                {rubricSamples}
            </div>
        );

        return sampleResponses;
    }

    render() {
        const label = `${this.props.language} Rubric`;
        return (
            <div className="item-information-content">
                {this.renderRubrics()}
                {this.renderSampleResponses()}
            </div>
        );
    }
}