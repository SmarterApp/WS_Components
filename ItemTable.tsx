import * as ItemCard from "./ItemCard";
import * as ItemCardViewModel from './ItemCardViewModel';
import * as React from "react";

export type Header = "Item" | "Claim/Target" | "Subject" | "Grade" | "Item Type";

export enum SortDirection {
    NoSort = 0,
    Ascending = 1,
    Descending = -1
}

export interface HeaderSort {
    col: SortColumn;
    direction: SortDirection;
    resetSortCount: number;
}

export interface SortColumn {
    header: Header;
    className: string;
    accessor: (label: ItemCardViewModel.ItemCardViewModel) => string | number;
    compare: (a: ItemCardViewModel.ItemCardViewModel, b: ItemCardViewModel.ItemCardViewModel) => number;
}

const invokeResetSortLimit = 2;

const decendingArrow = (
    <span aria-label="Decend">▼</span>
);

const acendingArrow = (
    <span aria-label="Ascend">▲</span>
);

const noSort = (
    <span aria-label="NoSort"></span>
);

export const headerColumns: SortColumn[] = [
    {
        header: "Item",
        className: "item",
        accessor: label => label.itemKey,
        compare: (a, b) => (a.itemKey) - (b.itemKey)
    },
    {
        header: "Claim/Target",
        className: "claimAndTarget",
        accessor: label => label.claimLabel + "/" + label.target,
        compare: (a, b) => {
            if (a.claimCode < b.claimCode || a.target < b.target) {
                return 1
            }
            else if (a.claimCode > b.claimCode || a.target > b.target) {
                return -1
            }
            else {
                return 0;
            }
        }
    },
    {
        header: "Subject",
        className: "subject",
        accessor: label => label.subjectLabel,
        compare: (a, b) => (a.subjectCode).localeCompare(b.subjectCode)
    },
    {
        header: "Grade",
        className: "grade",
        accessor: label => label.gradeLabel,
        compare: (a, b) => (a.grade) - (b.grade)
    },
    {
        header: "Item Type",
        className: "interactionType",
        accessor: label => label.interactionTypeLabel,
        compare: (a, b) => (a.interactionTypeCode).localeCompare(b.interactionTypeCode)
    },
];

interface HeaderProps {
    columns: SortColumn[];
    onHeaderClick: (header: SortColumn) => void;
    sorts: HeaderSort[];
}

export class HeaderTable extends React.Component<HeaderProps, {}> {
    constructor(props: HeaderProps){
        super(props);
    }

    compareColumn(lhs: ItemCardViewModel.ItemCardViewModel, rhs: ItemCardViewModel.ItemCardViewModel): number {
        const sorts = this.props.sorts || [];
        for (const sort of sorts) {
            const diff = sort.col.compare(lhs, rhs) * sort.direction;
            if (diff !== 0) {
                return diff;
            }
        }
        return 0;
    }

    headerEventHandler(scol: SortColumn, hcol: HeaderSort|undefined) {
        this.props.onHeaderClick(scol);
    }

    renderHeader(col: SortColumn): JSX.Element {
        let dirElem: string | undefined | JSX.Element;
        const headerSort = this.props.sorts.find(hs => hs.col.header === col.header);
        if (!headerSort) {
            dirElem = noSort;
        } else if (headerSort.direction === SortDirection.Ascending) {
            dirElem = acendingArrow;
        } else if (headerSort.direction === SortDirection.Descending) {
            dirElem = decendingArrow;
        } else {
            dirElem = noSort;
        }

        return (
            <th key={col.header}
                className={col.className}
                onClick={() => this.headerEventHandler(col,headerSort)}>
                <div className={col.className}>
                    {dirElem} {col.header}
                </div>
            </th>
        );
    }

    render() {
        return (
            <table className="item-table table mapcomponent-table">
                <thead>
                    <tr className="primary">
                        <th></th>
                        {this.props.columns.map(col => this.renderHeader(col))}
                    </tr>
                </thead>
            </table>
        );
    }
}

interface Props {
    tableRef?: (ref: HTMLTableElement) => void;
    mapRows: ItemCardViewModel.ItemCardViewModel[];
    rowOnClick: (item: ItemCardViewModel.ItemCardViewModel) => void;
    sort: HeaderSort[];
    columns: SortColumn[];
    selectedRow?: ItemCardViewModel.ItemCardViewModel;
}

export class DataTable extends React.Component<Props, {}> {
    constructor(props: Props) {
        super(props);
    }

   

    renderCell(col: SortColumn, cellData: ItemCardViewModel.ItemCardViewModel): JSX.Element {
        return (
            <td key={col.header}
                className={col.className}>
                <div className={col.className}>
                    {col.accessor(cellData)}
                </div>
            </td>
        );
    }

    //TODO replace X with checkmark icon 
    renderRow(rowData: ItemCardViewModel.ItemCardViewModel, index: number): JSX.Element {
        let isSelected = false;
        if (this.props.selectedRow) {
            isSelected = rowData.itemKey === this.props.selectedRow.itemKey
                        && rowData.bankKey === this.props.selectedRow.bankKey;
        }
        
        return (
            <tr key={index} className={isSelected ? "selected" : ""}
                onClick={() => {
                    this.props.rowOnClick(rowData);
                }}>
                <td>{isSelected ? "X" : ""}</td>
                {this.props.columns.map(col => this.renderCell(col, rowData))}
            </tr>
        );
    }

    renderRows(): JSX.Element {
        const rows = this.props.mapRows.map((rowData,idx) => this.renderRow(rowData,idx));
        return (
            <tbody>{rows}</tbody>
        );
    }

    render() {
        return (
            <table className="item-table table table-striped mapcomponent-table"
                ref={this.props.tableRef}>
                {this.renderRows()}
            </table>
        );
    }
}