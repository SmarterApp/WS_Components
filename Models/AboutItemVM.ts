import * as Rubric from '../PageTabs/Rubric'
import * as ItemCardViewModel from './ItemCardViewModel'
import * as API from '../Models/ApiModels'

export const ScoreSearchClient = (params: { bankKey: number, itemKey: number }) => API.get<AboutThisItem>("/api/aboutItem", params)

export interface AboutThisItem {
    rubrics: Rubric.Rubric[];
    itemCardViewModel: ItemCardViewModel.ItemCardViewModel;
    depthOfKnowledge: string;
    targetDescription: string;
    commonCoreStandardsDescription: string;
    educationalDifficulty: string;
    evidenceStatement: string;
}