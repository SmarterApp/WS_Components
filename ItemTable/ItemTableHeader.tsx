import * as ItemCardViewModel from '../Models/ItemCardViewModel';
import * as React from "react";

export type Header = "Item" | "Claim/Target" | "Subject" | "Grade" | "Item Type";

export interface Props {
    columns: SortColumn[];
    onHeaderClick: (header: SortColumn) => void;
    sorts: HeaderSort[];
}

export interface State {
    dirElem: string | undefined | JSX.Element;
}

export enum SortDirection {
    NoSort = 0,
    Ascending = 1,
    Descending = -1
}

export interface HeaderSort {
    col: SortColumn;
    direction: SortDirection;
    resetSortCount: number;
}

export interface SortColumn {
    header: Header;
    className: string;
    accessor: (label: ItemCardViewModel.ItemCardViewModel) => string | number;
    compare: (a: ItemCardViewModel.ItemCardViewModel, b: ItemCardViewModel.ItemCardViewModel) => number;
}

const invokeResetSortLimit = 2;

const decendingArrow = (
    <span aria-label="Decend">▼</span>
);

const acendingArrow = (
    <span aria-label="Ascend">▲</span>
);

const noSort = (
    <span aria-label="NoSort"></span>
);

export const headerColumns: SortColumn[] = [
    {
        header: "Item",
        className: "item",
        accessor: label => label.itemKey,
        compare: (a, b) => (a.itemKey) - (b.itemKey)
    },
    {
        header: "Claim/Target",
        className: "claimAndTarget",
        accessor: label => label.claimLabel + "/" + label.target,
        compare: (a, b) => {
            if (a.claimCode < b.claimCode || a.target < b.target) {
                return SortDirection.Ascending;
            }
            else if (a.claimCode > b.claimCode || a.target > b.target) {
                return SortDirection.Descending
            }
            else {
                return SortDirection.NoSort;
            }
        }
    },
    {
        header: "Subject",
        className: "subject",
        accessor: label => label.subjectLabel,
        compare: (a, b) => (a.subjectCode).localeCompare(b.subjectCode)
    },
    {
        header: "Grade",
        className: "grade",
        accessor: label => label.gradeLabel,
        compare: (a, b) => (a.grade) - (b.grade)
    },
    {
        header: "Item Type",
        className: "interactionType",
        accessor: label => label.interactionTypeLabel,
        compare: (a, b) => (a.interactionTypeCode).localeCompare(b.interactionTypeCode)
    },
];

export class HeaderTable extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = {
            dirElem: undefined
        }
    }

    compareColumn(lhs: ItemCardViewModel.ItemCardViewModel, rhs: ItemCardViewModel.ItemCardViewModel): number {
        const sorts = this.props.sorts || [];
        for (const sort of sorts) {
            const diff = sort.col.compare(lhs, rhs) * sort.direction;
            if (diff !== 0) {
                return diff;
            }
        }
        return 0;
    }

    headerEventHandler(scol: SortColumn, hcol: HeaderSort | undefined) {
        this.props.onHeaderClick(scol);
    }

    setDirElem(headerSort: HeaderSort) {
        let dirElem = undefined;
        if (!headerSort) {
            dirElem = noSort;
        } else if (headerSort.direction === SortDirection.Ascending) {
            dirElem = acendingArrow;
        } else if (headerSort.direction === SortDirection.Descending) {
            dirElem = decendingArrow;
        } else {
            dirElem = noSort;
        }

        this.setState({
            dirElem: dirElem
        })
    }

    renderHeader(col: SortColumn): JSX.Element {
        const headerSort = this.props.sorts.find(hs => hs.col.header === col.header);
        if (headerSort) {
            this.setDirElem(headerSort); 
        }

        return (
            <th key={col.header}
                className={col.className}
                onClick={() => this.headerEventHandler(col, headerSort)}>
                <div className={col.className}>
                    {this.state.dirElem} {col.header}
                </div>
            </th>
        );
    }

    render() {
        return (
            <table className="item-table table mapcomponent-table">
                <thead>
                    <tr className="primary">
                        {this.props.columns.map(col => this.renderHeader(col))}
                    </tr>
                </thead>
            </table>
        );
    }
}