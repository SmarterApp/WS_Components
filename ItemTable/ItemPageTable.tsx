import * as React from 'react'
import * as ItemTableHeader from './ItemTableHeader'
import * as ItemTable from './ItemTable'
import * as ItemCardViewModel from '../Models/ItemCardViewModel'
import * as GradeLevels from "../Models/GradeLevels";
import { ItemsSearchViewModel } from "../ScoreGuide/ItemSearchContainer";
import * as ItemModels from '../Models/ItemModels';
import * as ItemSearchDropdown from '../DropDown/ItemSearchDropdown';


export interface Props {
    onRowSelection: (item: {itemKey: number; bankKey: number}) => void;
    itemCards?: ItemCardViewModel.ItemCardViewModel[]; 
}

export interface State {
    sorts: ItemTableHeader.HeaderSort[]; 
    selectedRow?: ItemCardViewModel.ItemCardViewModel; 
    
}

export class ItemPageTable extends React.Component<Props, State>{
    private headerColumns = ItemTableHeader.headerColumns;
    private dataTableRef: HTMLTableElement;

    constructor(props: Props) {
        super(props); 
        this.state = {
            sorts: []
        }
    }

    onClickHeader = (col: ItemTableHeader.SortColumn) => {
        const newSorts = (this.state.sorts || []).slice();
        const headIdx = newSorts.findIndex(hs => hs.col.header === col.header);
        if (headIdx !== -1) {
            const newSort = Object.assign({}, newSorts[headIdx]);
            if (newSort.direction == ItemTableHeader.SortDirection.Ascending) {
                newSort.direction = ItemTableHeader.SortDirection.Descending;
            }
            else if (newSort.direction == ItemTableHeader.SortDirection.Descending) {
                newSort.direction = ItemTableHeader.SortDirection.NoSort;
            }
            else {
                newSort.direction = ItemTableHeader.SortDirection.Ascending;
            }

            newSorts[headIdx] = newSort;
        } else {
            const newSort: ItemTableHeader.HeaderSort = {
                col: col,
                direction: ItemTableHeader.SortDirection.Ascending,
                resetSortCount: 0
            };
            newSorts.push(newSort);
        }
        this.setState({ sorts: newSorts });
    }

    onSelectItem = (item: ItemCardViewModel.ItemCardViewModel) => {
        this.setState({
            selectedRow: item
        });
        const card = {itemKey: item.itemKey, bankKey: item.bankKey}

        this.props.onRowSelection(card);
     
    };

    invokeMultiSort(lhs: ItemCardViewModel.ItemCardViewModel, rhs: ItemCardViewModel.ItemCardViewModel): number {
        const sorts = this.state.sorts || [];
        for (const sort of sorts) {
            const diff = sort.col.compare(lhs, rhs) * sort.direction;
            if (diff !== 0) {
                return diff;
            }
        }
        return 0;
    }
    getTableData = (): ItemCardViewModel.ItemCardViewModel[] | undefined => {
        let data = this.props.itemCards; 
        if(data != undefined){
            data = this.state.sorts && this.state.sorts.length !== 0
                ? data.sort((lhs, rhs) => this.invokeMultiSort(lhs, rhs))
                : data;
        }
        return data; 
    }
    
    renderTableHeader() {
        return (
            <ItemTableHeader.HeaderTable
                sorts={this.state.sorts}
                onHeaderClick={this.onClickHeader}
                columns={this.headerColumns} />
        );
    }

    renderTable() {
        const itemCards = this.getTableData(); //this returns undefined but in the method it has data. that's w
        if (itemCards != undefined) {
            //if no items are returned we want to return a friendly message
            if (itemCards.length === 0) {
                return <span className="placeholder-text" role="alert">No results found for the given search terms.</span>
            } else {
                return (
                    <ItemTable.DataTable
                        mapRows={itemCards}
                        rowOnClick={this.onSelectItem}
                        sort={this.state.sorts}
                        tableRef={ref => this.dataTableRef = ref}
                        columns={this.headerColumns}
                        selectedRow={this.state.selectedRow} />
                );
            }
        }
        return <span className="placeholder-text" role="alert">No results found for the given search terms.</span>
    }

    render() {
        return (
            <div>
                {this.renderTableHeader()};
                {this.renderTable()};
            </div>
        );
    }
}
