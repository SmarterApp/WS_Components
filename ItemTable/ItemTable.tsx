import * as ItemCardViewModel from '../Models/ItemCardViewModel';
import * as React from "react";
import * as ItemTableHeader from './ItemTableHeader';
import * as AboutItemVM from '../Models/AboutItemVM';
import * as ApiModels from '../Models/ApiModels';
import * as ItemCardViewer from '../AboutItem/ItemCardViewer';

interface Props {
    tableRef?: (ref: HTMLTableElement) => void;
    mapRows: ItemCardViewModel.ItemCardViewModel[];
    rowOnClick: (item: ItemCardViewModel.ItemCardViewModel) => void;
    sort: ItemTableHeader.HeaderSort[];
    columns: ItemTableHeader.SortColumn[];
    selectedRow?: ItemCardViewModel.ItemCardViewModel;
    item: ApiModels.Resource<AboutItemVM.AboutThisItem>;
}

export class DataTable extends React.Component<Props, {}> {
    constructor(props: Props) {
        super(props);
    }

    renderCell(col: ItemTableHeader.SortColumn, cellData: ItemCardViewModel.ItemCardViewModel): JSX.Element {
        return (
            <td key={col.header}
                className={col.className}>
                <div className={col.className}>
                    {col.accessor(cellData)}
                </div>
            </td>
        );
    }

    //TODO replace X with a > that specifies that  the table  row can  be expanded
    renderRow(rowData: ItemCardViewModel.ItemCardViewModel, index: number, isSelected?: boolean): JSX.Element {
        const style = { color: "gray" }
        const collapse = (<i style={style} className="fa fa-chevron-right fa-sm" aria-hidden="true"></i>)
        const expand = (<i style={style} className="fa fa-chevron-down fa-sm" aria-hidden="true"></i>)
        let tab = null;

        return (
            <tr key={index} className={isSelected ? "selected" : ""}
                onClick={() => {
                    this.props.rowOnClick(rowData);
                }}>
                <td>{isSelected ? expand : collapse}</td>
                {this.props.columns.map(col => this.renderCell(col, rowData))}
            </tr>
        );
    }

    renderTabsContainer() {
        let content = null;
        if (this.props.item.kind == "success" || this.props.item.kind == "reloading") {
            const newItem = this.props.item.content;
            return (
                <div>
                    <ItemCardViewer.ItemCardViewer
                        item={newItem}
                    />
                </div>
            );

        }
        return content;
    }

    renderRows(): JSX.Element[] {
        let rows: JSX.Element[] = [];
        if (this.props.item.kind === "success" || this.props.item.kind === "reloading") {
            const mapRows = this.props.mapRows;
            let isSelected = false;
            for (let i = 0; i < mapRows.length + 1; i++) {
                if (this.props.selectedRow) {
                    isSelected = mapRows[i].itemKey === this.props.selectedRow.itemKey
                        && mapRows[i].bankKey === this.props.selectedRow.bankKey;
                    rows.push(this.renderRow(mapRows[i], i, isSelected))
                    if (isSelected) {
                        rows.push(
                            <tr>
                                <td colSpan={6}>
                                    <ItemCardViewer.ItemCardViewer
                                        item={this.props.item.content}
                                    />
                                </td>
                            </tr>
                        )
                    }
                    i++;
                }
                else {
                    rows.push(this.renderRow(mapRows[i], i, isSelected))
                }

            }
        } else {
            rows = this.props.mapRows.map((rowData, idx) => this.renderRow(rowData, idx, false))
        }
        return rows;

    }

    render() {
        return (<tbody>{this.renderRows()}</tbody>);
    }
}