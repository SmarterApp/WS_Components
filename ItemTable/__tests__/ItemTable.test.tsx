import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { DataTable } from '../ItemTable';
import * as TestUtils from 'react-dom/test-utils';
import * as ItemCardViewModel from '../../Models/ItemCardViewModel';
import * as GradeLevels from '../../Models/GradeLevels';
import * as Rubric from '../../PageTabs/Rubric';
import * as AboutItemVM from '../../Models/AboutItemVM';
import * as ApiModels from '../../Models/ApiModels';
import { shallow, mount } from 'enzyme';

describe("ItemTable", () => {
    const itemCards: ItemCardViewModel.ItemCardViewModel[] = [{
        bankKey: 1,
        itemKey: 3,
        title: "",
        grade: GradeLevels.GradeLevels.All,
        gradeLabel: "",
        subjectCode: "",
        subjectLabel: "",
        claimCode: "",
        claimLabel: "",
        target: "",
        interactionTypeCode: "",
        interactionTypeLabel: "",
        isPerformanceItem: true
    }]

    const rubrics: Rubric.Rubric[] = []

    const content: AboutItemVM.AboutThisItem = {
        rubrics,
        itemCardViewModel: itemCards[0],
        depthOfKnowledge: "depthOfKnowledge",
        targetDescription: "targetDescription",
        commonCoreStandardsDescription: "commonCoreStandardsDescription",
        educationalDifficulty: "educationalDifficulty",
        evidenceStatement: "evidenceStatement"
    }

    const item: ApiModels.Resource<AboutItemVM.AboutThisItem> = {
        kind: "none",
        content
    }
    
    const props = {
        tableRef: jest.fn(),
        mapRows: [],
        rowOnClick: jest.fn(),
        sort: [],
        columns: [],
        selectedRow: {
                bankKey: 1,
                itemKey: 123,
                title: "test_title",
                grade: 1,
                gradeLabel: "1",
                subjectCode: "Math",
                subjectLabel: "MTH",
                claimCode: "claim_1234",
                claimLabel: "claim_math",
                target: "",
                interactionTypeCode: "type01",
                interactionTypeLabel: "type01label",
                isPerformanceItem: false
        },
        item
    }
    
    it("matches snapshot", () => {
        expect(shallow(<DataTable {...props}/>)).toMatchSnapshot();;
    })
})