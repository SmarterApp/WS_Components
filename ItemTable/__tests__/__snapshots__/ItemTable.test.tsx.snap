// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemTable matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <tbody>
    
</tbody>,
  "nodes": Array [
    <tbody>
      
</tbody>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <DataTable
        columns={Array []}
        mapRows={Array []}
        rowOnClick={[Function]}
        selectedRow={
                Object {
                        "bankKey": 1,
                        "claimCode": "claim_1234",
                        "claimLabel": "claim_math",
                        "grade": 1,
                        "gradeLabel": "1",
                        "interactionTypeCode": "type01",
                        "interactionTypeLabel": "type01label",
                        "isPerformanceItem": false,
                        "itemKey": 123,
                        "subjectCode": "Math",
                        "subjectLabel": "MTH",
                        "target": "",
                        "title": "test_title",
                      }
        }
        sort={Array []}
        tableRef={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": DataTable {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "columns": Array [],
          "mapRows": Array [],
          "rowOnClick": [Function],
          "selectedRow": Object {
            "bankKey": 1,
            "claimCode": "claim_1234",
            "claimLabel": "claim_math",
            "grade": 1,
            "gradeLabel": "1",
            "interactionTypeCode": "type01",
            "interactionTypeLabel": "type01label",
            "isPerformanceItem": false,
            "itemKey": 123,
            "subjectCode": "Math",
            "subjectLabel": "MTH",
            "target": "",
            "title": "test_title",
          },
          "sort": Array [],
          "tableRef": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <tbody>
          
</tbody>,
        "_debugID": 2,
        "_renderedOutput": <tbody>
          
</tbody>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DataTable
    columns={Array []}
    mapRows={Array []}
    rowOnClick={[Function]}
    selectedRow={
        Object {
            "bankKey": 1,
            "claimCode": "claim_1234",
            "claimLabel": "claim_math",
            "grade": 1,
            "gradeLabel": "1",
            "interactionTypeCode": "type01",
            "interactionTypeLabel": "type01label",
            "isPerformanceItem": false,
            "itemKey": 123,
            "subjectCode": "Math",
            "subjectLabel": "MTH",
            "target": "",
            "title": "test_title",
          }
    }
    sort={Array []}
    tableRef={[Function]}
/>,
}
`;
